operator/controller/dnsnameresolver/resolver.go

```diff
@@ -233,9 +235,12 @@ func (resolver *Resolver) Delete(objName string) {
 		delete(resolver.wildcardObjInfo, objName)
 		matchesRegular = false
 	}
+	resolver.dnsLock.Unlock()
 
 	// Iterate through the dnsNameList slice.
 	for _, dnsName := range dnsNameList {
+		resolver.dnsLock.Lock()
```

Comment:
We need to release the lock before sending the signal on the delete channel. That's why I have unlocked before the loop and then taken the lock inside it. Probably the lock can be released before sending the signal to the channel and then again take the lock on return.

Reply:
Otherwise, we can create another slice of DNS names which needs sending signal to the delete channel. Once the changes have been made to the `resolver.dnsNames` then we can release the lock and send the signals one by one in a loop.